packages/nextjs/src/server/content-security-policy.ts

```diff
@@ -0,0 +1,357 @@
+   * @param values - Array of values to process
+   * @returns Set of formatted values
+   */
+  static handleDirectiveValues(values: string[]): Set<string> {
+    const result = new Set<string>();
+
+    if (values.includes("'none'") || values.includes('none')) {
+      result.add("'none'");
+      return result;
+    }
+
+    values.forEach(v => result.add(this.formatValue(v)));
+    return result;
+  }
+}
+
+/**
+ * Handles merging of existing directives with new values
+ * @param mergedCSP - The current merged CSP state
+ * @param key - The directive key to handle
+ * @param values - New values to merge
+ */
+function handleExistingDirective(mergedCSP: CSPDirectiveSet, key: CSPDirective, values: string[]) {
+  // Special case for 'none' value - it should replace all other values
+  if (values.includes("'none'") || values.includes('none')) {
+    mergedCSP[key] = new Set(["'none'"]);
+    return;
+  }
+
+  // For existing directives, merge the values rather than replacing
+  // Use a Set to deduplicate values
+  const deduplicatedSet = new Set<string>();
+
+  // First add existing values after formatting them
+  mergedCSP[key].forEach(value => {
+    deduplicatedSet.add(CSPDirectiveManager.formatValue(value));
+  });
+
+  // Then add new values after formatting them
+  values.forEach(value => {
+    deduplicatedSet.add(CSPDirectiveManager.formatValue(value));
+  });
+
+  // If this is script-src in production, make sure we don't add unsafe-eval
+  if (key === 'script-src' && process.env.NODE_ENV === 'production') {
+    deduplicatedSet.delete("'unsafe-eval'");
+  }
+
+  mergedCSP[key] = deduplicatedSet;
+}
+
+/**
+ * Handles custom directives that are not part of the default set
+ * @param customDirectives - Map of custom directives
+ * @param key - The directive key
+ * @param values - Values for the directive
+ */
+function handleCustomDirective(customDirectives: Map<string, Set<string>>, key: string, values: string[]) {
+  // Special case for 'none' value - it should replace all other values
+  if (values.includes("'none'") || values.includes('none')) {
+    customDirectives.set(key, new Set(["'none'"]));
+    return;
+  }
+
+  // For all other cases, create a new set with formatted values
+  const formattedValues = new Set<string>();
+  values.forEach(value => {
+    const formattedValue = CSPDirectiveManager.formatValue(value);
+    formattedValues.add(formattedValue);
+  });
+
+  customDirectives.set(key, formattedValues);
+}
+
+/**
+ * Formats the CSP header string with proper ordering and formatting
+ * @param mergedCSP - The merged CSP state to format
+ * @returns Formatted CSP header string
+ */
+function formatCSPHeader(mergedCSP: Record<string, Set<string>>): string {
+  const orderMap: Record<string, number> = {
+    "'none'": 1,
+    "'self'": 2,
+    "'strict-dynamic'": 3,
+    "'unsafe-eval'": 4,
+    "'unsafe-inline'": 5,
+    'http:': 6,
+    'https:': 7,
+  };
```

Comment:
❓ why is ordering important here?

Reply:
It's not. More of an OCD thing

Reply:
I can rip that out to keep things simpler

Reply:
Done